renderdoc/core/resource_manager.h

```diff
@@ -675,6 +706,19 @@ void ResourceManager<Configuration>::MarkResourceFrameReferenced(ResourceId id,
   if(id == ResourceId())
     return;
 
+  if(IsDirtyFrameRef(refType))
+  {
+    // If the resource was postponed during Active Capture, we need to prepare it
+    // right away, since next Read might be invalid.
```

Comment:
I've realised there's another case where we need to do a last-minute prepare: if the resource is deleted and is about to go away, but was perhaps used as read-only within the frame.

Reply:
OK, what would be the correct place to do this prepare?

Reply:
I think from looking that we always unregister the resource *before* deleting the actual API object, so it should be safe to do the prepare in `ReleaseCurrentResource`. At least that's true for `glDeleteBuffers` and `glDeleteTextures` which should be enough.